mqtt/mqtt-broker/src/broker.rs

```diff
@@ -454,6 +454,7 @@ where
 
             // Ungraceful disconnect - send the will
             if let Some(will) = session.into_will() {
```

Comment:
should we make a new `fn try_send_will(&self, session)` that will contain logic and logging?